classes/class-kco-checkout.php

```diff
@@ -108,15 +108,24 @@ public function update_klarna_order() {
 		}
 
 		$klarna_order = KCO_WC()->api->get_klarna_order( $klarna_order_id );
+		if ( ! $klarna_order ) {
+			KCO_Logger::log( "Klarna order could not be retrieved during update for ID: $klarna_order_id " );
+			return;
+		}
 
-		if ( $klarna_order && 'checkout_incomplete' === $klarna_order['status'] ) {
+		if ( 'checkout_incomplete' === $klarna_order['status'] ) {
 			// If it is, update order.
-			$klarna_order = KCO_WC()->api->update_klarna_order( $klarna_order_id );
+			$updated_klarna_order = KCO_WC()->api->update_klarna_order( $klarna_order_id );
+			if ( ! $updated_klarna_order ) {
+				KCO_Logger::log( "Klarna order could not be updated during update for ID: $klarna_order_id " );
+				return;
+			}
 		}
 
 		// If cart doesn't need payment anymore - reload the checkout page.
 		if ( apply_filters( 'kco_check_if_needs_payment', true ) ) {
-			if ( ! WC()->cart->needs_payment() && 'checkout_incomplete' === $klarna_order['status'] ) {
+			$status = $updated_klarna_order ? $updated_klarna_order['status'] : $klarna_order['status'];
```

Comment:
Issue: `$updated_klarna_order` isn't set in all cases here. So we should probably set it before the if check on line 116 to false by default if we don't make the request.